src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerTest.java

```diff
@@ -58,25 +66,26 @@
   private DatasetReportService datasetReportService;
 
   @Test
-  void processDataset_expectSuccess() throws Exception {
+  void processDatasetFromURL_expectSuccess() throws Exception {
 
     var dataset = new MockMultipartFile("dataset", "dataset.txt", "text/plain",
-        "<test></test>".getBytes());
+            "<test></test>".getBytes());
 
     var records = List.of(new ByteArrayInputStream("record1".getBytes()),
-        new ByteArrayInputStream("record2".getBytes()));
+            new ByteArrayInputStream("record2".getBytes()));
 
     var datasetObject = new Dataset("12345", Set.of(), 0);
 
-    when(zipService.parse(dataset)).thenReturn(records);
+    when(harvestService.harvest(dataset)).thenReturn(records);
     when(datasetService.createDataset("my-data-set", ITALY, IT, records)).thenReturn(datasetObject);
 
     mvc.perform(multipart("/dataset/{name}/process", "my-data-set")
-        .file(dataset)
-        .param("country", ITALY.name())
-        .param("language", IT.name()))
-        .andExpect(status().isAccepted())
-        .andExpect(jsonPath("$.dataset-id", is("12345")));
+                    .file(dataset)
+                    .param("country", ITALY.name())
+                    .param("language", IT.name())
+                    .param("URL", ""))
```

Comment:
The URL parameter doesn't seem to be needed here.
Same goes for all methods below..